metadata/content.go

```diff
@@ -367,16 +441,62 @@ func (cs *contentStore) checkAccess(ctx context.Context, dgst digest.Digest) err
 }
 
 func readInfo(info *content.Info, bkt *bolt.Bucket) error {
-	return bkt.ForEach(func(k, v []byte) error {
-		switch string(k) {
-		case string(bucketKeyCreatedAt):
-			if err := info.CommittedAt.UnmarshalBinary(v); err != nil {
-				return err
-			}
-		case string(bucketKeySize):
-			info.Size, _ = binary.Varint(v)
+	if err := readTimestamps(&info.CommittedAt, &info.UpdatedAt, bkt); err != nil {
+		return err
+	}
+
+	lbkt := bkt.Bucket(bucketKeyLabels)
+	if lbkt != nil {
+		info.Labels = map[string]string{}
+		if err := readLabels(info.Labels, lbkt); err != nil {
+			return err
 		}
-		// TODO: Read labels
-		return nil
+	}
+
+	if v := bkt.Get(bucketKeySize); len(v) > 0 {
+		info.Size, _ = binary.Varint(v)
+	}
+
+	return nil
+}
+
+func writeInfo(info *content.Info, bkt *bolt.Bucket) error {
+	if err := writeTimestamps(bkt, info.CommittedAt, info.UpdatedAt); err != nil {
+		return err
+	}
+
+	if err := writeLabels(bkt, info.Labels); err != nil {
+		return errors.Wrapf(err, "writing labels for info %v", info.Digest)
+	}
+
+	// Write size
+	sizeEncoded, err := encodeSize(info.Size)
+	if err != nil {
+		return err
+	}
+
+	if err := bkt.Put(bucketKeySize, sizeEncoded); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func adaptContentInfo(info content.Info) filters.Adaptor {
```

Comment:
Place this in `adaptors.go` for now. Having these next to each will make it easier to maintain consistency and merge common patterns.